### PR TITLE
Developer Documentation Integration

### PR DESCRIPTION
**Pull request by chrisdennis**
So this is a very coarse cut of integrating the existing developer documentation in to the main repository.  What I'm mainly interested in at the moment, is people's opinions on which of these files should be left in the wiki?

### CODE REVIEW

File: docs/src/docs/asciidoc/developer/dev.git.asciidoc [hunk 1]
```diff
@@ -0,0 +1,124 @@
+= Git usage
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/developer/dev.git.asciidoc]: Wiki

File: docs/src/docs/asciidoc/developer/dev.guidelines.asciidoc [hunk 2]
```diff
@@ -0,0 +1,112 @@
+= Developer guidelines
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/developer/dev.guidelines.asciidoc]: Wiki

File: docs/src/docs/asciidoc/developer/dev.meetings.asciidoc [hunk 3]
```diff
@@ -0,0 +1,294 @@
+= Meet up notes
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/developer/dev.meetings.asciidoc]: Delete

File: docs/src/docs/asciidoc/developer/module.api.asciidoc [hunk 4]
```diff
@@ -0,0 +1,33 @@
+= API Module
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/developer/module.api.asciidoc]: Yes. Should be here.

File: docs/src/docs/asciidoc/developer/module.clustering.asciidoc [hunk 5]
```diff
@@ -0,0 +1,69 @@
+= Terracotta's Clustering Module
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/developer/module.clustering.asciidoc]: Here

File: docs/src/docs/asciidoc/developer/module.core.asciidoc [hunk 6]
```diff
@@ -0,0 +1,39 @@
+= Core Module
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/developer/module.core.asciidoc]: Here

File: docs/src/docs/asciidoc/developer/module.impl.asciidoc [hunk 7]
```diff
@@ -0,0 +1,23 @@
+= Implementation Module
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/developer/module.impl.asciidoc]: Here

File: docs/src/docs/asciidoc/developer/module.xa.asciidoc [hunk 8]
```diff
@@ -0,0 +1,60 @@
+= Transactions module
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/developer/module.xa.asciidoc]: Here. And a lot of modules are missing

**Comment by chrisdennis** [comment]
Will implement Henri's suggestions and re-push.

**Comment by chrisdennis** [comment]
Looking at the content of the wiki I'm thinking the sensible split is to put design and code implementation docs in the main repo, and leave process related documentation in the wiki.  Thats what this latest version of the PR attempts.

**Comment by henri-tremblay** [comment]
Yes. I fully agree with that

**Pull request merged**